includes/components/class-attribute.php

```diff
@@ -307,6 +310,72 @@ public function register_models() {
 		}
 	}
 
+	/**
+	 * Validates unique model attributes.
+	 *
+	 * @param array  $errors Error messages.
+	 * @param object $model Model object.
+	 * @return array
+	 */
+	public function validate_unique_attributes( $errors, $model ) {
+		if ( empty( $errors ) ) {
+
+			// Set unique fields.
+			$fields = [];
+
+			foreach ( $model->_get_fields() as $field_name => $field ) {
+				if ( ! hp\get_array_value( $field->get_args(), 'unique' ) || ! $field->get_value() ) {
+					continue;
+				}
+
+				$fields[] = [
+					'name'  => $field_name,
+					'value' => $field->get_value(),
+					'label' => $field->get_label(),
+				];
+			}
+
+			// Check fields.
+			if ( ! $fields ) {
+				return $errors;
+			}
+
+			// Get model class.
+			$model_class = hp\create_class_instance( '\HivePress\Models\\' . $model::_get_meta( 'name' ) );
+
+			// Check model class.
+			if ( ! $model_class ) {
+				return $errors;
+			}
+
+			foreach ( $fields as $field ) {
+
+				// Get models.
+				$models = $model_class->query()->filter(
+					[
+						'status__in' => [ 'auto-draft', 'draft', 'pending', 'publish' ],
+						'id__not_in' => [ $model->get_id() ],
+					]
+				)->set_args(
+					[
+						'meta_key'   => hp\prefix( $field['name'] ),
+						'meta_value' => $field['value'],
+					]
+				)->get_count();
```

Comment:
Count is a bit expensive, we need to know if at least 1 model has the same value so you can use get_first_id